### DISCUSSION TITLE
export SerializeFrom

### DISCUSSION THREAD

**Discussion by yoshizawa56** [remix-run/react-router#12417] [Proposals]
## Background
With the integration of Remix into React Router v7, the `SerializeFrom` type that was previously exported from Remix is no longer available. This type was particularly useful for type definitions when working with loader data.

## Use Case
The `SerializeFrom` type was helpful in properly typing the data returned from loaders, ensuring type safety when accessing loader data in components. For example:
```typescript
// Before (in Remix)
import type { SerializeFrom, LoaderFunction } from "@remix-run/node";

const loader: LoaderFunction = async () => {
  return json({ user: { id: 1, name: "John" } });
};

type LoaderData = SerializeFrom<typeof loader>;
```

## Request
Could you consider re-exporting the `SerializeFrom` type in React Router v7? This would help maintain consistency for projects migrating from Remix and provide a useful typing utility for the broader React Router community.

Thank you for considering this request.

**Comment by vincaslt** [comment]
With react-router v7 it seems you don't need to use `json` and can return a plain object, then use this:

```
type SerializeFrom<T> = ReturnType<typeof useLoaderData<T>>;
```

see: https://remix.run/docs/en/main/start/future-flags#serializefrom

**Reply by yoshizawa56** [reply]
Thank you for your response! 
The type utility `SerializeFrom<T> = ReturnType<typeof useLoaderData<T>>` seems applicable to my use case as well.
